e2e/e2e.go

```diff
@@ -0,0 +1,240 @@
+// Package e2e sets up end to end tests with a stubbed IPVS so it can run in build environments.
+// The main purpose is to test the client/server -> store CRUD functionality.
+// The actual specs are located in api/ and meradm/.
+package e2e
+
+import (
+	"os"
+
+	"io/ioutil"
+
+	"os/exec"
+
+	"bytes"
+	"strings"
+
+	"fmt"
+
+	"strconv"
+
+	"crypto/sha256"
+	"io"
+	"net"
+	"net/http"
+	"syscall"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	etcdDownloadURL    = "https://github.com/coreos/etcd/releases/download/v2.3.8/etcd-v2.3.8-linux-amd64.tar.gz"
+	etcdExpandedPath   = "etcd-v2.3.8-linux-amd64/"
+	etcdDownloadSha256 = "3431112f97105733aabb95816df99e44beabed2cc96201679e3b3b830ac35282"
+	// assumes all specs run in subfolders of e2e
+	workingDir  = "../.."
+	buildDir    = workingDir + "/build"
+	etcdBinary  = buildDir + "/etcd"
+	etcdTarball = buildDir + "/etcd.tar.gz"
+)
+
+var (
+	etcdListenPort   string
+	etcdPeerPort     string
+	merlinPort       string
+	merlinHealthPort string
+	dataDir          string
+	etcd             *exec.Cmd
+	merlin           *exec.Cmd
+)
+
+func MeradmList() []string {
+	return strings.Split(strings.TrimSpace(Meradm("list")), "\n")
+}
+
+func Meradm(args ...string) string {
+	out, err := Meradm2(args...)
+	Expect(err).ToNot(HaveOccurred())
+	return out
+}
+
+func Meradm2(args ...string) (string, error) {
```

Comment:
Maybe a little renaming to make the purpose clearer? How about calling this one `Meradm` and the function above `MeradmWithNoErrors`.